File: python/leetcode/203. Remove Linked List Elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None


class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        if head is None:
            return head
        nxt = self.removeElements(head.next, val)
        if head.val == val:
            head = nxt
        else:
            head.next = nxt

        return head

File: go/leetcode/203. Remove Linked List Elements.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeElements(head *ListNode, val int) *ListNode {
    if head == nil {return head}
    
    nxt := removeElements(head.Next, val)
    if head.Val == val {
        head = nxt
    } else {
        head.Next = nxt
    }
    return head
}
  